.github/workflows/android_builds.yml

```diff
@@ -78,6 +78,7 @@ jobs:
           scons-cache-limit: ${{ matrix.cache-limit }}
 
       - name: Save Godot build cache
+        if: always()
```

Comment:
This makes it run even if previous steps fail, which ensures that if the cache was locked, it gets unlocked.

If the job fails before `cache/restore`, then this step still runs and passes fine with just warnings:
```
Warning: Path Validation Error: Path(s) specified in the action for caching do(es) not exist, hence no cache is being saved.
Warning: Cache save failed.
```

Reply:
What I haven't tested though is what happens if the job is _canceled_ (e.g. by pushing a new commit to the PR branch). Let's test this in this PR and makes sure it also runs this steps and unlocks the cache key.